rslib/i18n_helpers/src/garbage_collection.rs

```diff
@@ -0,0 +1,222 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+use std::{collections::HashSet, fs, io::BufReader};
+
+use fluent_syntax::{ast, parser};
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde_json;
+use walkdir::{DirEntry, WalkDir};
+
+use crate::serialize;
+
+/// Extract references from all Rust, Python, TS, Svelte and Designer files in
+/// the `roots`, convert them to kebab case and write them as a json to the
+/// target file.
+pub fn extract_ftl_references(roots: &[&str], target: &str) {
+    let mut refs = HashSet::new();
+    for root in roots {
+        for entry in WalkDir::new(root) {
+            let entry = entry.expect("failed to visit dir");
+            if entry.file_type().is_file() {
+                extract_references_from_file(&mut refs, &entry);
+            }
+        }
+    }
+    serde_json::to_writer(
+        fs::File::create(target).expect("failed to create file"),
+        &refs,
+    )
+    .expect("failed to write file");
```

Comment:
I didn't implement any error handling and the `expect`s will need to be replaced depending on how you plan to use these scripts.

Reply:
For build scripts, tests, and tools like these, unwrap and expect are fine; I'd likely do the same